src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -671,6 +679,99 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
     );
   }
 
+  showDuplicateModal = () => {
+    this.setState({ isShowingDuplicateModal: true });
+  };
+
+  hideDuplicateModal = () => {
+    this.setState({ isShowingDuplicateModal: false });
+  };
+
+  renderDuplicateModal() {
+    const { workspaces, http, notifications } = this.props;
+    const { isShowingDuplicateModal, duplicateSelectedSavedObjects, duplicateMode } = this.state;
+
+    if (!isShowingDuplicateModal) {
+      return null;
+    }
+
+    const onDuplicate = async (
+      savedObjects: SavedObjectWithMetadata[],
+      includeReferencesDeep: boolean,
+      targetWorkspace: string
+    ) => {
+      const objectsToDuplicate = savedObjects.map((obj) => ({ id: obj.id, type: obj.type }));
+      let result;
+      try {
+        result = await duplicateSavedObjects(
+          http,
+          objectsToDuplicate,
+          includeReferencesDeep,
+          targetWorkspace
+        );
+        if (result.success) {
+          notifications.toasts.addSuccess({
+            title: i18n.translate(
+              'savedObjectsManagement.objectsTable.duplicate.successNotification',
+              {
+                defaultMessage:
+                  'Duplicate ' + savedObjects.length.toString() + ' saved objects successfully',
+              }
+            ),
+          });
+        } else if (result.errors) {
+          const errorsIds = result.errors.map((item: { id: string }) => item.id);
+          notifications.toasts.addDanger({
+            title: i18n.translate(
+              'savedObjectsManagement.objectsTable.duplicate.dangerNotification',
+              {
+                defaultMessage:
+                  'Unable to duplicate ' +
+                  savedObjects.length.toString() +
+                  ' saved objects. These objects cannot be duplicated:' +
+                  errorsIds.join(','),
+              }
+            ),
+          });
+        } else {
+          notifications.toasts.addDanger({
+            title: i18n.translate(
+              'savedObjectsManagement.objectsTable.duplicate.dangerNotification',
+              {
+                defaultMessage:
+                  'Unable to duplicate ' + savedObjects.length.toString() + ' saved objects',
+              }
+            ),
+          });
```

Comment:
Can we combine these two `else` into one?

Reply:
Yes, I will do this.